src/cocoa/toga_cocoa/widgets/button.py

```diff
@@ -1,67 +1,39 @@
 from rubicon.objc import objc_method, get_selector
-
-from toga.interface import Button as ButtonInterface
-from toga.constants import *
-
-from .base import WidgetMixin
+from .base import Widget
 from ..libs import *
 from ..utils import process_callback
 
 
 class TogaButton(NSButton):
     @objc_method
     def onPress_(self, obj) -> None:
-        if self._interface.on_press:
-            process_callback(self._interface.on_press(self._interface))
-
+        if self.interface.on_press:
+            process_callback(self.interface.on_press(self.interface))
 
-class Button(ButtonInterface, WidgetMixin):
-    def __init__(self, label, id=None, style=None, on_press=None, enabled=None,
-                background_color=None):
-        super().__init__(label, id=id, style=style, on_press=on_press,
-                        enabled=enabled, background_color=background_color)
-        self._create()
 
+class Button(Widget):
     def create(self):
-        self._impl = TogaButton.alloc().init()
-        self._impl._interface = self
+        self.native = TogaButton.alloc().init()
+        self.native.interface = self.interface
 
-        self._impl.setBezelStyle_(NSRoundedBezelStyle)
-        self._impl.setButtonType_(NSMomentaryPushInButton)
-        self._impl.setTarget_(self._impl)
-        self._impl.setAction_(get_selector('onPress:'))
+        self.native.setBezelStyle_(NSRoundedBezelStyle)
+        self.native.setButtonType_(NSMomentaryPushInButton)
+        self.native.setTarget_(self.native)
+        self.native.setAction_(get_selector('onPress:'))
 
         # Add the layout constraints
-        self._add_constraints()
+        self.add_constraints()
 
-    def _set_label(self, label):
-        self._impl.setTitle_(self.label)
+    def set_label(self, label):
+        self.native.setTitle_(label)
         self.rehint()
 
-    def _set_enabled(self, value):
-        self._impl.setEnabled_(self.enabled)
-
-    def _set_background_color(self, background_color):
```

Comment:
I moved it into cocoa.Widget. We talked about it and came to the conclusion that every Widget should have the ability to be enabled and disabled.